CONTRIBUTING.md

```diff
@@ -1,92 +1,146 @@
-all managed by [Bundler](http://bundler.io/) according to the
-[Puppet support matrix](http://docs.puppetlabs.com/guides/platforms.html#ruby-versions).
+```bash
+bundle exec pdk
+```
 
-By default the tests use a baseline version of Puppet.
+Also static and style checks are now possible.  For instance checking the puppet
+and ruby code can be done like this:
 
-If you have Ruby 2.x or want a specific version of Puppet,
-you must set an environment variable such as:
+```bash
+bundle exec rake markdownlint validate lint check rubocop
+```
 
-    export PUPPET_VERSION="~> 3.2.0"
+Checking the Markdown files with
+[markdownlint](https://github.com/markdownlint/markdownlint) can be done like
+this:
 
-Install the dependencies like so...
+```bash
+bundle exec rake markdownlint
+```
 
-    bundle install
+## Unit test
 
-## Syntax and style
+The unit tests are implemented in `spec/unit/` and can be called by
+running:
 
-The test suite will run [Puppet Lint](http://puppet-lint.com/) and
-[Puppet Syntax](https://github.com/gds-operations/puppet-syntax) to
-check various syntax and style things. You can run these locally with:
+```bash
+bundle exec rake spec
+```
 
-    bundle exec rake lint
-    bundle exec rake syntax
+For parallel execution of the unit tests run:
 
-## Running the unit tests
+```bash
+bundle exec rake parallel_spec
+```
 
-The unit test suite covers most of the code, as mentioned above please
-add tests if you're adding new functionality. If you've not used
-[rspec-puppet](http://rspec-puppet.com/) before then feel free to ask
-about how best to test your new feature. Running the test suite is done
-with:
+## Acceptance tests
 
-    bundle exec rake spec
+The unit tests just check if the code runs and contains the required resource
+definitions.  In those tests, the code is not used to provision real systems.
+For this [litmus](https://github.com/puppetlabs/puppet_litmus) is used.
 
-Note also you can run the syntax, style and unit tests in one go with:
+Litmus fires up fresh containers or virtual machines, applies this puppet
+module and then checks if it behaved as expected by running a series of tests
+on it. Those tests are implemented in `spec/acceptance/`.
 
-    bundle exec rake test
+Litmus is normally run in a Github action on each pull request.  But it is also
+possible to run it on a local machine.  The following examples use the Docker
+provisioner to provision a Debian 11 system with Puppet 7.  For other options
+see the [litmus documentation](https://puppetlabs.github.io/litmus/) and
+the list of [available images](https://github.com/puppetlabs/litmusimage).
 
-### Automatically run the tests
+First, a machine has to be provisioned:
 
-During development of your puppet module you might want to run your unit
-tests a couple of times. You can use the following command to automate
-running the unit tests on every change made in the manifests folder.
+```bash
+bundle exec rake 'litmus:provision[docker,litmusimage/debian:11]'
```

Comment:
This fails, when I try this locally. Looking at the logs of the Docker container yields this:
```shell-session
$ docker logs 547d731f041e
Failed to look up module alias 'autofs4': Function not implemented
Failed to mount cgroup at /sys/fs/cgroup/systemd: No such file or directory
[!!!!!!] Failed to mount API filesystems.
Exiting PID 1...
```

Reply:
I will add a line to inform devs that if it doesn't work on their system locally, they should just use the github actions runners in a draft pull request.